### ISSUE TITLE
Cannot load enzyme library

### ISSUE THREAD

**Issue by m-mujica** [stealjs/steal#680]
I've set up an example app that uses https://github.com/airbnb/enzyme but I can't get it to work

1. Steal throws trying to load enzyme's [main](https://github.com/airbnb/enzyme/blob/603f07adb31032d9b917b49685276f39390cfb76/package.json#L5) which points to the build folder where there is an `index.js` (which should be loaded by default in a CJS environment). My workaround was to use a combination of [map/paths](https://github.com/m-mujica/enzyme-steal-example/blob/cd136a73053f1566b7e3f108e18545fc34ce59ee/package.json#L20) to make Steal load the right file. Should steal try to load `index.js` if `build/build` is missing?

2. Enzyme depends on cheerio which tries to load its `package.json` using `require('./package')`, I guess that the extension in this case is required by the new spec, right? fwiw, the cheerio folks [changed that](https://github.com/cheeriojs/cheerio/commit/7b59afbc7a6aa39376021593e011bf718350bcb4) but it's still not available in npm. The workaround in this case was to add a [map using the full module name](https://github.com/m-mujica/enzyme-steal-example/blob/cd136a73053f1566b7e3f108e18545fc34ce59ee/package.json#L16)

3. Once the previous errors were fixed, steal throws trying to load a couple of deps (`react/lib/ReactContext`, `react/lib/ExecutionEnvironment`....), the modules are indeed missing, looking at [enzyme's docs](https://github.com/airbnb/enzyme/blob/master/docs/guides/webpack.md) I found that for webpack they set these modules as `externals`; from the webpack docs:

  ```
  **externals** allows you to specify dependencies for your library 
  that are not resolved by webpack, but become dependencies of the output.
  This means they are imported from the environment during runtime.
  ```

  I think steal does not have support for this (browserify supports this as well). Should we add support   for `externals`?

4. Just for the sake of testing, I [mapped the missing modules of 3 to `@empty` ](https://github.com/m-mujica/enzyme-steal-example/blob/cd136a73053f1566b7e3f108e18545fc34ce59ee/package.json#L14-L15) and got a new round of errors :( 

<img width="1433" alt="screen shot 2016-05-03 at 21 32 14" src="https://cloud.githubusercontent.com/assets/724877/15013127/fb9272f0-11d3-11e6-8e64-5739d036437f.png">

  I just looked that the fist one there, I'm not sure why steal is trying to load `object-keys/package.json` from within `define-properties/node_modules` when `object-keys` is a sibling of `define-properties`.

Here is the example app code https://github.com/m-mujica/enzyme-steal-example, What do you guys think?

**Comment by matthewp** [comment]
Couple of things after a quick look:

1. Turn on npm3 support (if you're using it) with `npmAlgorithm: "flat"`
2. `npm install steal-builtins --save`  to make the use of util, events, etc. work. These are node builtin modules.

**Comment by m-mujica** [comment]
@matthewp 

1. I'm on node@4.4.3, npm@2.15.1
2. I tried that it fixed some errors, I updated the example app code. 

![screen shot 2016-05-04 at 09 00 15](https://cloud.githubusercontent.com/assets/724877/15013647/46e07196-11d7-11e6-909b-41fffc86bf4e.png)

**Comment by matthewp** [comment]
One problem is that there is some code like this:

```
var EVENTS = require("../").EVENTS;
```

We should make that be `../index`.

**Comment by matthewp** [comment]
Adding this before parsing the moduleName in steal-extension.js fixes that part:

```
if(name === '../') {
  name = name + 'index';
}
```

**Comment by matthewp** [comment]
Another thing that the same piece of code is doing about is when it does:

```
require('../');
```

There is not index, so what is happening is, because this is the parent most folder, it is looking in package.json and loading the main.  We need to support that with retries.

**Comment by matthewp** [comment]
https://github.com/stealjs/system-npm/issues/133

**Comment by matthewp** [comment]
This should be fixed in 0.16.5 but I haven't tried the example app yet.

**Comment by matthewp** [comment]
This is fixed, you can even remove the package.json map!
